modules/promotions/assets/js/react/behavior.js

```diff
@@ -0,0 +1,63 @@
+import App from './app';
+import { createRoot } from 'react-dom/client';
+
+export default class ReactPromotionBehavior extends Marionette.Behavior {
+	promotionInfoTip = null;
+
+	selectors = {
+		reactAnchor: '.e-promotion-react',
+	};
+
+	ui() {
+		return {
+			animatedHeadlineButton: '[data-promotion].elementor-control-type-switcher',
+		};
+	}
+
+	events() {
+		return {
+			'click @ui.animatedHeadlineButton': 'onClickControlButtonAnimatedHeadline',
+		};
+	}
+
+	onClickControlButtonAnimatedHeadline( event ) {
+		event.stopPropagation();
+		this.mount();
+	}
+
+	mount() {
+		const colorScheme = elementor?.getPreferences?.( 'ui_theme' ) || 'auto',
+			isRTL = elementorCommon.config.isRTL;
+
+		const rootElement = document.querySelector( this.selectors.reactAnchor );
+
+		if ( ! rootElement ) {
+			return;
+		}
+
+		if ( this.promotionInfoTip ) {
```

Comment:
this check can be moved to the top, so that nothing is executed while into tip is already shown....